l10n_br_nfse/models/document.py

```diff
@@ -0,0 +1,177 @@
+    )
+    rps_type = fields.Selection(
+        string='RPS Type',
+        selection=RPS_TYPE,
+        default='1',
+    )
+    operation_nature = fields.Selection(
+        string='Operation Nature',
+        selection=OPERATION_NATURE,
+        default='1',
+    )
+    taxation_special_regime = fields.Selection(
+        string='Taxation Special Regime',
+        selection=TAXATION_SPECIAL_REGIME,
+        default='1',
+    )
+    verify_code = fields.Char(
+        string='Verify Code',
+        readonly=True,
+    )
+    nfse_environment = fields.Selection(
+        selection=NFSE_ENVIRONMENTS,
+        string="NFSe Environment",
+        default=lambda self: self.env.user.company_id.nfse_environment,
+    )
+
+    def document_number(self):
+        for record in self.filtered(fiter_processador_edoc_nfse):
+            if record.issuer == DOCUMENT_ISSUER_COMPANY:
+                if record.document_serie_id:
+                    record.document_serie = record.document_serie_id.code
+                    if not record.rps_number and record.date:
+                        record.rps_number = record.document_serie_id.\
+                            next_seq_number()
+        super(Document, self).document_number()
+
+    def _generate_key(self):
+        remaining = self - self.filtered(fiter_processador_edoc_nfse)
+        if remaining:
+            super(Document, remaining)._generate_key()
+
+    def _processador_erpbrasil_nfse(self):
+        certificado = cert.Certificado(
+            arquivo=self.company_id.certificate_nfe_id.file,
+            senha=self.company_id.certificate_nfe_id.password
+        )
+        session = Session()
+        session.verify = False
+        transmissao = TransmissaoSOAP(certificado, session)
+        return NFSeFactory(
+            transmissao=transmissao,
+            ambiente=self.nfse_environment,
+            cidade_ibge=int('%s%s' % (
+                self.company_id.partner_id.state_id.ibge_code,
+                self.company_id.partner_id.city_id.ibge_code
+            )),
+            cnpj_prestador=misc.punctuation_rm(
+                self.company_id.partner_id.cnpj_cpf),
+            im_prestador=misc.punctuation_rm(
+                self.company_id.partner_id.inscr_mun or ''),
+        )
+
+    @api.multi
+    def _document_export(self, pretty_print=True):
+        super(Document, self)._document_export()
+        for record in self.filtered(fiter_processador_edoc_nfse):
+            edoc = record.serialize()[0]
+            processador = record._processador_erpbrasil_nfse()
+            xml_file = processador.\
+                _generateds_to_string_etree(edoc, pretty_print=pretty_print)[0]
+            event_id = self._gerar_evento(xml_file, event_type="0")
+            _logger.debug(xml_file)
+            record.autorizacao_event_id = event_id
+
+    def _prepare_dados_servico(self):
+        self.line_ids.ensure_one()
+        result = {
+            'codigo_cnae': None,
```

Comment:
Pelo que entendi esta linha está fixando o codigo_cnae como vazio certo? No caso da ISSNet se faz necessário  informar o codigo_cnae no XML. O que pode ser feito aqui para não afetar nem a ginfes e nem issnet ?

@mileo @luismalta

Reply:
```suggestion
            'codigo_cnae': None,
```

Ao revisar o PR que o @marcelsavegnago  fez no @odoo-brazil comentei com o @luismalta que a implementação correta implementar um campo cnae no cadastro de produtos, não obrigatório, e carrega-lo neste dict.

Achei inclusive que ele já tinha feito isso.

Reply:
@mileo Como sugestão, este campo poderia estar relacionado com os cnaes configurados na empresa. O que acha ?